lib/ansible/modules/windows/win_get_url.ps1

```diff
@@ -69,41 +88,40 @@ Function Download-File($result, $url, $dest, $headers, $credentials, $timeout, $
     }
 
     # TODO: Replace this with WebRequest
-    $webClient = New-Object System.Net.WebClient
+    $extWebClient = New-Object ExtendedWebClient
 
     foreach ($header in $headers.GetEnumerator()) {
-        $webClient.Headers.Add($header.Name, $header.Value)
+        $extWebClient.Headers.Add($header.Name, $header.Value)
     }
 
-# FIXME: WebClient has no Timeout property ? Should be replaced with WebRequest
-#    if ($timeout) {
-#        $webClient.Timeout = $timeout * 1000
-#    }
+    if ($timeout) {
+        $extWebClient.Timeout = $timeout * 1000
+    }
 
     if (-not $use_proxy) {
         # Ignore the system proxy settings
-        $webClient.Proxy = $null
+        $extWebClient.Proxy = $null
     } elseif ($proxy) {
-        $webClient.Proxy = $proxy
+        $extWebClient.Proxy = $proxy
     }
 
     if ($credentials) {
-        $webClient.Credentials = $credentials
+        $extWebClient.Credentials = $credentials
     }
 
-    Try {
-        if (-not $whatif) {
-            $webClient.DownloadFile($url, $dest)
+    if (-not $whatif) {
+        Try {
+            $extWebClient.DownloadFile($url, $dest)
+        } Catch [System.Net.WebException] {
+            $result.status_code = $_.Exception.Response.StatusCode
```

Comment:
You would have to try and cast this as an `[int]`, this returns the human readable enum value and not the number.